testonly/hammer/hammer.go

```diff
@@ -85,6 +85,34 @@ const (
 
 var mapEntrypoints = []MapEntrypointName{GetLeavesName, SetLeavesName, GetSMRName, GetSMRRevName}
 
+// Constants for both valid and invalid operation choices.
+type Choice string
+const (
+	ExistingKey    = Choice("ExistingKey")
+	NonexistentKey = Choice("NonexistentKey")
+	MalformedKey   = Choice("MalformedKey")
+	DuplicateKey   = Choice("DuplicateKey")
+	RevTooBig      = Choice("RevTooBig")
+	RevIsZero      = Choice("RevIsZero")
+	RevIsNegative  = Choice("RevIsNegative")
+	CreateLeaf     = Choice("CreateLeaf")
+	UpdateLeaf     = Choice("UpdateLeaf")
+	DeleteLeaf     = Choice("DeleteLeaf")
+)
+
+type choicesByEntrypoint map[MapEntrypointName][]Choice
+
+var validChoicesByEntrypoint = choicesByEntrypoint{
+	GetLeavesName: []Choice{ExistingKey, NonexistentKey},
+	SetLeavesName: []Choice{CreateLeaf, UpdateLeaf, DeleteLeaf},
+}
+
+var invalidChoicesByEntrypoint = choicesByEntrypoint{
+	GetLeavesName: []Choice{MalformedKey, RevTooBig, RevIsZero}, // TODO(phad): RevIsNegative ought to be invalid too but is accepted atm.
+	SetLeavesName: []Choice{MalformedKey, DuplicateKey},
+	GetSMRRevName: []Choice{RevTooBig, RevIsZero, RevIsNegative},
+}
+
```

Comment:
I much prefer the previous structure of having the relevant options visible within the function that uses them, so it's immediately clear which options are available and that the `switch` does indeed cover all of them.

Reply:
Happy to move those back.
We then dispense with a map lookup-by-string-key which it occurred to me might be a waste of CPU cycles when (I assume) you want the hammer to spin as fast as it can.